numpy/core/src/multiarray/arraytypes.c.src

```diff
@@ -1119,7 +1129,15 @@ static void
     @totype@ *op = output;
 
     while (n--) {
-        *op++ = (@totype@)*ip;
+        @fromtype@ f = *ip++;
```

Comment:
This is a bug, this line should read `*ip` - the `++` happens on line 1141

Reply:
Fixed and added more tests to cover this.